capture_service/device_mgr_test.cc

```diff
@@ -17,19 +17,329 @@ limitations under the License.
 #include "device_mgr.h"
 
 #include "absl/status/status.h"
+#include "absl/status/status_matchers.h"
+#include "gmock/gmock.h"
 #include "gtest/gtest.h"
 
 namespace Dive
 {
-
 namespace
 {
 
+using ::absl_testing::IsOkAndHolds;
+using ::absl_testing::StatusIs;
+
+MATCHER_P(GfxrReplaySettingsEq, expected, "")
+{
+    EXPECT_EQ(arg.remote_capture_path, expected.remote_capture_path);
+    EXPECT_EQ(arg.local_download_dir, expected.local_download_dir);
+    EXPECT_EQ(arg.run_type, expected.run_type);
+    EXPECT_EQ(arg.metrics, expected.metrics);
+    EXPECT_EQ(arg.loop_single_frame_count, expected.loop_single_frame_count);
+    return true;
+}
+
+TEST(ValidateGfxrReplaySettingsTest, NoLocalDownloadDirFail)
+{
+    GfxrReplaySettings rs = {};
+    rs.remote_capture_path = "PLACEHOLDER_REMOTE_CAPTURE_PATH";
+
+    absl::StatusOr<GfxrReplaySettings> ret = ValidateGfxrReplaySettings(rs, /*is_adreno_gpu=*/true);
+    EXPECT_THAT(ret,
+                StatusIs(absl::StatusCode::kInvalidArgument, "Must provide local_download_dir"))
+    << ret.status();
```

Comment:
nit: For StatusOr, I think that this can be slightly implied by checking `.status()` instead of the StatusOr. That would reduce the need to explicitly print the status and would allow you to inline the function call for better test output:
```suggestion
    EXPECT_THAT(ValidateGfxrReplaySettings(rs, /*is_adreno_gpu=*/true).status(),
                StatusIs(absl::StatusCode::kInvalidArgument, "Must provide local_download_dir"));
```